# DSCI 100 Project

In [2]:
library(repr)
library(tidyverse)
library(tidymodels)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

## Introduction

We wish to help Prof. Frank to figure out how to what types of players will be more likely to contribute more data.

There's a reddit rumor saying that truly dedicated people don't stop, never stop. They have obsessions. For example, mathematicans may spend years studying a single problem, while most people would not persist for years. Moderators on reddit may spends hours each day moderating posts on reddit, despite they are unpaid.

This leads me to think: are true gamers sleepless? If a gamer plays game at 4 AM, will the gamer be a real gamer that would provide endless data?

In this project, we will explore whether participants who play game overnight will have more `played_hours`.

## Filter players by whether they play overnight

To answer the question, the very first step we need to do is to remove all records consisting of players who do not stay overnight. We say a player stays overnight to play game if the player plays game between 0 AM to 4 AM.

Let's start with loading the datasets.

In [7]:
players <- read_csv("data/players.csv") |> mutate(gender = as.factor(gender))
sessions <- read_csv("data/sessions.csv")
players
sessions

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


ERROR: Error in players.head(4): could not find function "players.head"


Then, let's proceed to to parse the start time and the end time.

In [11]:
sessions_with_parsed_time <- sessions |> mutate(start_time = as_datetime(start_time, format = "%d/%m/%Y %H:%M"), 
                                                end_time = as_datetime(end_time, format = "%d/%m/%Y %H:%M"))
sessions_with_parsed_time

hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7    fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8    ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9    96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
11   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
12   f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
13   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
14   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
15   24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
16   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
17   88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
18   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
19   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
20   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
21   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
22   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
23   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
24   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
25   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
26   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
27   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
28   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
29   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
30   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
⋮    ⋮                                                               
1506 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
1507 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1508 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1509 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
1510 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1511 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1512 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1513 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1514 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1515 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1516 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1517 a1e0438d568b9c05a93cadaf99969e4d44794d19dea31edd723cbc916f878c33
1518 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1519 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
1520 55d24216db39c27e1f17cc43d3127cbf8ed76ada6d098202b53ded319855e2c1
1521 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1522 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1523 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1524 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1525 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1526 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1527 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1528 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1529 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1530 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1531 36d9cbb4c6

Then, we proceed to extract out the hours and minutes of start time and end time, and convert the hours to to minutes. I read https://ubc-dsci.github.io/dsci-100-student/REFERENCE_R.html over and over again and didn't see how to get hours and minutes out of a `dttm`. Therefore, we have to go out of the box and use `format`, which is not part of reference sheet.

In [16]:
sessions_with_time_in_hr <- sessions_with_parsed_time |> mutate(
    start_time_in_hr = as.numeric(format(start_time, "%H"))
    + as.numeric(format(start_time, "%M")) / 60,
    end_time_in_hr = as.numeric(format(end_time, "%H")) 
    + as.numeric(format(end_time, "%M")) / 60,
  )
sessions_with_time_in_hr

hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7    fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8    ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9    96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
11   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
12   f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
13   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
14   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
15   24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
16   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
17   88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
18   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
19   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
20   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
21   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
22   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
23   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
24   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
25   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
26   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
27   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
28   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
29   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
30   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
⋮    ⋮                                                               
1506 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
1507 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1508 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1509 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
1510 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1511 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1512 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1513 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1514 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1515 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1516 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1517 a1e0438d568b9c05a93cadaf99969e4d44794d19dea31edd723cbc916f878c33
1518 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1519 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
1520 55d24216db39c27e1f17cc43d3127cbf8ed76ada6d098202b53ded319855e2c1
1521 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1522 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1523 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1524 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1525 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1526 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1527 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1528 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1529 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1530 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1531 36d9cbb4c6

Now, we have start time and end time in hour, it suffices for us to use filter. For a player to play overnight, there are following cases:
- Start time is between 0 AM to 4 AM
- Start time is greater than the end time (the player played pass midnight)

In [18]:
filtered_sessions <- sessions_with_time_in_hr |> filter(start_time_in_hr < 4 | start_time_in_hr > end_time_in_hr)
filtered_sessions

hashedEmail                                                     
1   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
3   96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
4   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
5   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
6   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
7   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
8   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
9   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
10  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
11  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
12  fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
13  fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
14  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
15  fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
16  36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
17  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
18  ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
19  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
20  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
21  b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
22  ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
23  fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
24  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
25  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
26  f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
27  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
28  ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
29  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
30  42eafe96ed5c1684e3b5cc614d1b01a117173d3ec6898abcf1177d9822bd39f7
⋮   ⋮                                                               
553 bc704ff2bc676dbf48ee41b9e11481c1387bf758ad318f2428f336e3fecc6660
554 0088b5e134c3f0498a18c7ea6b8d77b4b0ff1636fc93355ccc95b45423367832
555 a1e0438d568b9c05a93cadaf99969e4d44794d19dea31edd723cbc916f878c33
556 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
557 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
558 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
559 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
560 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
561 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
562 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
563 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
564 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
565 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
566 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
567 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
568 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
569 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
570 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
571 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
572 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
573 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
574 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
575 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
576 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
577 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
578 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
57

Now, suffices to use semi_join to join `filtered_sessions` with `players`, to filter all players who play overnight.

In [20]:
filtered_players <- players |> semi_join(filtered_sessions, by = "hashedEmail")
filtered_players

experience subscribe
1  Pro         TRUE    
2  Veteran     TRUE    
3  Amateur     TRUE    
4  Regular     TRUE    
5  Veteran     TRUE    
6  Amateur     TRUE    
7  Amateur     TRUE    
8  Regular     TRUE    
9  Amateur     TRUE    
10 Veteran     TRUE    
11 Veteran     TRUE    
12 Veteran    FALSE    
13 Regular     TRUE    
14 Veteran     TRUE    
15 Veteran     TRUE    
16 Veteran     TRUE    
17 Beginner    TRUE    
18 Regular     TRUE    
19 Amateur     TRUE    
20 Regular     TRUE    
21 Regular     TRUE    
22 Pro         TRUE    
23 Veteran     TRUE    
24 Amateur     TRUE    
25 Regular     TRUE    
26 Amateur     TRUE    
27 Regular     TRUE    
28 Amateur     TRUE    
29 Beginner    TRUE    
30 Amateur     TRUE    
31 Beginner    TRUE    
32 Amateur     TRUE    
33 Beginner    TRUE    
34 Veteran    FALSE    
35 Veteran     TRUE    
36 Beginner   FALSE    
37 Beginner    TRUE    
38 Beginner   FALSE    
39 Regular     TRUE    
40 Amateur     TRUE    
41 Beginner    TRUE    
42 Beginner    TRUE    
43 Amateur     TRUE    
44 Amateur     TRUE    
45 Regular     TRUE    
46 Beginner   FALSE    
47 Veteran     TRUE    
48 Beginner   FALSE    
49 Veteran    FALSE    
50 Amateur     TRUE    
51 Veteran     TRUE    
52 Amateur    FALSE    
   hashedEmail                                                     
1  f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2  f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3  23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5
4  7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e
5  7a4686586d290c67179275c7c3dfb4ea02f4d317d9ee0e2cee98baa27877a875
6  a175d4741dc84e6baf77901f6e8e0a06f54809a34e6b5211159bced346f7fb3e
7  ab1f44f93c3b828f55458971db393052d9711df3e0e7ff69540bfebfcec555ff
8  bc704ff2bc676dbf48ee41b9e11481c1387bf758ad318f2428f336e3fecc6660
9  4b01bce3f141289709e8278b02ba5d2aaa7105d7ccb9c7deb37670a80e332774
10 f1b432523542f90c61176a555ccb2144468d76c91a32d74082ab8c101f9d25b6
11 5a340c0e3d1aa3e579efc625bd3e5bca7fc25f7115b68e9411f457464bfd3df4
12 88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
13 0d4d71be33e2bc7266ee4983002bd930f69d304288a8663529c875f40f1750f3
14 d07befbac08be3d876d026ac0458651067001b39081d24724af3c95be76355e0
15 8d2eed1f399e0d77cebb8fcc48ed19ad2fa8e3bb3fa683cb90e0ac6d42677bf2
16 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
17 d13020f8d53c08c95e64f2254424a65c9c479fcfd866b21b850d200a2689a8b5
18 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
19 8e0aac3020b3fd9cdef4840b533b4b105aaf1ce1f6f2df3f328ba2a44cecfdc1
20 f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
21 0088b5e134c3f0498a18c7ea6b8d77b4b0ff1636fc93355ccc95b45423367832
22 90f1495942837b1cde67cc9e3119421e38183502a4c6dee304df3bf0c21e0dd0
23 dc73467f73263dd4a07838330dd1cc115aa3f8b03538911ee6aac048405939c4
24 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
25 71e724778a28bb0ebbdacb1328944dfef427525466b012b66e7554a65e401a16
26 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
27 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
28 95aa10ff79645508dcff5c08da02a8f6ddd9016b44fb469347e5739b8865c613
29 d48302dee689942b60f7e9756eba99d0be030df0ab6f553904a9dbbc44d0b5d5
30 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
31 e3175c379bbb83c8786d2dd719cf844575aca24d2c81d643dc7f12f6227becf7
32 bd10e7eab5531d5e82c9eebb929e2703f05367a90b0533074392a5f39a55c5d9
33 96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
34 c871393f513c68e313874ac9d192a0b8be7bf99a96bff0ab46beb343ee4217d5
35 e44041459da2102dc20147ed6f0db4753547be66fc4ddea037bafca64004de2d
36 722ab6d91280c439f2780767a1250889e93f75fc0ba0bc48175da022ab3e9ea9
37 9c2a5275ada716508a7e66629dfbeaa4c03ca9eb5f30d2132f41de130c367c56
38 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
39 c57fa94e436cf49a929d0168e47d26fec3d900b321775e280ef136979c01d5a4
40 fd6563a4

Now, let us compute the average played hours across players who play overnight compared to the entire population, to see if the mean changes significantly.

In [24]:
mean_played_hour_overnight <- filtered_players |> summarize(mean = mean(played_hours))
mean_played_hour_overnight

mean_played_overall <- players |> summarize(mean = mean(played_hours))
mean_played_overall

mean
1 21.4

mean    
1 5.845918

As we can see, there's a significant difference! The mean played hour across players who played overnight is a whopping 21.4, while the mean played hour across all players is only 5.84!